core/eth/reader.go

```diff
@@ -654,44 +657,151 @@ func (t *Reader) GetAllVersionedBlobParams(ctx context.Context) (map[uint8]*core
 	return res, nil
 }
 
-func (t *Reader) GetActiveReservations(ctx context.Context, blockNumber uint32, accountIDs []string) (map[string]core.ActiveReservation, error) {
-	// contract is not implemented yet
-	return map[string]core.ActiveReservation{}, nil
+func (t *Reader) GetActiveReservations(ctx context.Context, accountIDs []gethcommon.Address) (map[gethcommon.Address]*core.ActiveReservation, error) {
+	if t.bindings.PaymentVault == nil {
+		return nil, errors.New("payment vault not deployed")
+	}
+	reservationsMap := make(map[gethcommon.Address]*core.ActiveReservation)
+	reservations, err := t.bindings.PaymentVault.GetReservations(&bind.CallOpts{
+		Context: ctx,
+	}, accountIDs)
+	if err != nil {
+		return nil, err
+	}
+
+	// since reservations are returned in the same order as the accountIDs, we can directly map them
+	for i, reservation := range reservations {
+		if isZeroValuedReservation(reservation) {
+			delete(reservationsMap, accountIDs[i])
```

Comment:
If a reservation is zero, this delete would be a no-op and it would be added in L677, no? 
Don't we want `if !isZeroValuedReservation(reservation) { reservationsMap[accountIDs[i]] = &reservation }` ?